system/system_windows.go

```diff
@@ -117,9 +157,23 @@ func Install(dir string, ps *goolib.PkgSpec) error {
 
 // Uninstall performs a system specfic uninstall given a packages PackageState.
 func Uninstall(dir string, ps *goolib.PkgSpec) error {
+	var s string
 	un := ps.Uninstall
+	// Automatically determine uninstall script if none is specified in spec.
 	if un.Path == "" {
-		return nil
+		switch filepath.Ext(ps.Install.Path) {
+		case ".msi":
+			u := uninstallString(dir, "msi")
+			u = strings.ReplaceAll(u, `/I`, `/X`)
+			commands := strings.Split(u, " ")
+			un.Path = commands[0]
+			un.Args = commands[1:]
+			un.Args = append([]string{"/qn", "/norestart"}, un.Args...)
```

Comment:
Is this double-applying the /qn and /norestart args from line 197?

Reply:
No, reason being is that the file name is msiexec.exe. This would enter the exe case(since registry is typically stored as msiexe.exe /x{guid_of_application}), where we need to pass all the args. The msi case would be used when passing the msi installer directly.